### PR TITLE
[Consignments] Move some tests to jest

### PR DESCRIPTION
**Pull request by damassi**
This PR moves some of our tests over to jest and typescript:
- Components 
- Analytics 

I used a [jest codemod](https://github.com/skovhus/jest-codemods) to do most of it. 

Got hung up on some mocha tests so will gradually migrate this to completion.

### CODE REVIEW

File: src/desktop/apps/consign/components/choose_artist/__tests__/ChooseArtist.jest.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    initialStore = createStore(reducers)
+  })
+
+  it("disables the button when there is no input", () => {
+    const wrapper = mount(<ChooseArtist store={initialStore} />)
+    expect(
+      wrapper
+        .find(".consignments-submission-choose-artist__next-button[disabled]")
+        .props().disabled
+    ).toBe(true)
```
Comment (pepopowitz) [inline src/desktop/apps/consign/components/choose_artist/__tests__/ChooseArtist.jest.tsx:21]:
```suggestion
    expect(
      wrapper
        .find(".consignments-submission-choose-artist__next-button[disabled]")
        .length
    ).toEqual(1)
```

The codemod translated these assertions a bit funny, I think. Line 19 says "give me the button that is disabled", and line 20/21 say "make sure the button that is disabled is disabled". 

I am not totally certain on the syntax of my suggestion, but I think it'd be worth updating these to assertions that don't unnecessarily inspect the `props` bag.

File: src/desktop/apps/consign/components/choose_artist/__tests__/ChooseArtist.jest.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  })
+
+  it("enables the button when there is an input", () => {
+    initialStore.dispatch(actions.updateArtistAutocompleteValue("andy"))
+    const wrapper = mount(<ChooseArtist store={initialStore} />)
+    expect(
+      wrapper
+        .find(".consignments-submission-choose-artist__next-button[disabled]")
+        .props().disabled
+    ).toBe(false)
```
Comment (pepopowitz) [inline src/desktop/apps/consign/components/choose_artist/__tests__/ChooseArtist.jest.tsx:31]:
```suggestion
    expect(
      wrapper
        .find(".consignments-submission-choose-artist__next-button[disabled]")
        .length
    ).toEqual(1)
```

File: src/desktop/apps/consign/components/upload_photo/__tests__/UploadPhoto.jest.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        actions.addImageToUploadedImages("astronaut.jpg", "blahblabhalbhablah")
+      )
+      const wrapper = mount(<UploadPhoto store={initialStore} />)
+      expect(
+        wrapper
+          .find(
+            ".consignments-submission-upload-photo__submit-button[disabled]"
+          )
+          .prop("disabled")
+      ).toBe(false)
```
Comment (pepopowitz) [inline src/desktop/apps/consign/components/upload_photo/__tests__/UploadPhoto.jest.tsx:43]:
```suggestion
      expect(
        wrapper
          .find(
            ".consignments-submission-upload-photo__submit-button[disabled]"
          )
          .length
      ).toEqual(0)
```

Strange that the codemod translated the assertion on line 24-28 one way, and the rest this way.

File: src/desktop/apps/consign/components/upload_photo/__tests__/UploadPhoto.jest.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        actions.addImageToUploadedImages("astronaut.jpg", "blahblabhalbhablah")
+      )
+      initialStore.dispatch(actions.startProcessingImage("astronaut.jpg"))
+      expect(
+        wrapper
+          .find(
+            ".consignments-submission-upload-photo__submit-button[disabled]"
+          )
+          .prop("disabled")
+      ).toBe(true)
```
Comment (pepopowitz) [inline src/desktop/apps/consign/components/upload_photo/__tests__/UploadPhoto.jest.tsx:59]:
```suggestion
      expect(
        wrapper
          .find(
            ".consignments-submission-upload-photo__submit-button[disabled]"
          )
          .length
      ).toEqual(1)
```

File: src/desktop/apps/consign/components/choose_artist/__tests__/ChooseArtist.jest.tsx (hunk 1 above)
Comment (damassi) [reply src/desktop/apps/consign/components/choose_artist/__tests__/ChooseArtist.jest.tsx:21]: This wasn't a codemod but an issue with Enzyme and perhaps how i swapped `shallow` for `render`: https://github.com/enzymejs/enzyme/issues/336. It's pretty minor and the overall behavior is still being checked. (This applies to all the following tests.)

**Pull request merged**